Kitodo/src/main/java/org/kitodo/production/services/workflow/WorkflowControllerService.java

```diff
@@ -399,12 +406,8 @@ public void solveProblem(Comment comment, TaskEditType taskEditType)
             taskService.save(currentTask);
         }
         comment.setCurrentTask(ServiceManager.getTaskService().getById(comment.getCurrentTask().getId()));
-        comment.setCorrected(Boolean.TRUE);
-        comment.setCorrectionDate(new Date());
-        try {
-            ServiceManager.getCommentService().saveToDatabase(comment);
-        } catch (DAOException e) {
-            Helper.setErrorMessage("errorSaving", new Object[] {"comment"}, logger, e);
+        if (Objects.nonNull(comment.getProcess())) {
+            comment.setProcess(ServiceManager.getProcessService().getById(comment.getProcess().getId()));
```

Comment:
We are setting the currentTask and the process again but do not save the comment object. If calling those setters is necessary should we not persist the comment object again?

Reply:
Yes, you are right, this is quite confusing. The reason why we re-add the process (and task!) to the comment without saving it again is that we are making use of a side effect here. This `Comment` object is used again in the calling method afterwards, but with an updated `currentTask` and `process` property. It's not a _different_ process or task that is added here, but rather _updated_ objects, so re-setting these properties in the comment acts as a "refresh" of the object. Since the database entry of the comment only contains the _ID_ of the `process` and the `currentTask`, there is no need to save it again, because the IDs didn't change (e.g. its the same process and currentTask as before, like I mentioned above).

This is indeed implemented in a sub-optimal way and should be refactored more fundamentally when we find the time to do so but right now I do not have the resources for that, unfortunately. I think the bug should be fixed anyway, even if its just an incremental improvement over the status quo and not yet the optimal solution. I can add a comment or `TODO` as a reminder to the code that this code should be refactored and improved, if you like.

Reply:
@solth I see, thanks a lot for the explanation. In think you can go ahead and merge. Adding a TODO would be helpful.